relay-server/src/actors/spooler/mod.rs

```diff
@@ -271,7 +285,8 @@ impl BufferService {
                 .max_connections(config.spool_envelopes_max_connections())
                 .min_connections(config.spool_envelopes_min_connections())
                 .connect_with(options)
-                .await?;
+                .await
+                .map_err(BufferError::SetupFailed)?;
```

Comment:
Could `setup()` return the `db`? Then this second intitialization would not be necessary. Or do we need different options here than in `setup()`?

Reply:
yeah, so we need different options for migrations and then usage. And migration must run in separate dedicated pool before everything else, to make sure that all the things setup properly.